packages/core/src/connectors/index.ts

```diff
@@ -33,6 +33,42 @@ export const getConnectorInstanceById = async (id: string): Promise<ConnectorIns
   return { connector, ...found };
 };
 
+const isSmsConnector = (connector: ConnectorInstance): connector is SmsConnector => {
+  return connector.metadata.type === ConnectorType.SMS;
+};
+
+export const getSmsConnectorInstance = async (): Promise<SmsConnector> => {
+  const connectors = await getConnectorInstances();
+  const smsConnector = connectors
+    .filter((connector) => connector.connector?.enabled)
+    .find((connector) => isSmsConnector(connector));
+
+  // Need to call `isSmsConnector` again, otherwise typescript won't change type for it.
+  if (!smsConnector || !isSmsConnector(smsConnector)) {
+    throw new Error('Can not find any available SmsConnector');
```

Comment:
ConnectoeError? Or will refactor later

Reply:
same, this should be 500 (normal error)

passwordless-related functions should not be enabled in admin console unless connectors are set properly.

Reply:
updated